core/rawdb/freezer.go

```diff
@@ -671,7 +429,7 @@ func (f *freezer) MigrateTable(kind string, convert convertLegacyFn) error {
 	// Set up new dir for the migrated table, the content of which
 	// we'll at the end move over to the ancients dir.
 	migrationPath := filepath.Join(ancientsPath, "migration")
-	newTable, err := NewFreezerTable(migrationPath, kind, FreezerNoSnappy[kind], false)
+	newTable, err := NewFreezerTable(migrationPath, kind, table.noCompression, false)
```

Comment:
Why did you change this? Is this necessarily correct?

Reply:
Firstly, `table. noCompression ` is initialized with `FreezerNoSnappy[kind]`, which is semantically correct.

And my reason for this change is: we can make freezer more self-contained. Later we can define more freezer instances(e.g. reverse diff freezer) and the compression info doesn't have to be defined in  `FreezerNoSnappy`(actually in ReverseDiffNoSnappy).

Reply:
Ah right, I sloppily read this as initializing it with "no compression", but you're just using whatever the table already is set to. LGTM